sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridExecutionUtils.scala

```diff
@@ -0,0 +1,262 @@
+      classOf[MicrosToTimestamp],
+      classOf[MillisToTimestamp],
+      classOf[Minute],
+      classOf[MonotonicallyIncreasingID],
+      classOf[Month],
+      classOf[NaNvl],
+      classOf[NextDay],
+      classOf[Not],
+      classOf[Or],
+      classOf[Overlay],
+      classOf[Pi],
+      classOf[Pow],
+      classOf[Quarter],
+      classOf[Rand],
+      classOf[Remainder],
+      classOf[Reverse],
+      classOf[Rint],
+      classOf[Round],
+      classOf[Second],
+      classOf[Sha1],
+      classOf[Sha2],
+      classOf[ShiftLeft],
+      classOf[ShiftRight],
+      classOf[Shuffle],
+      classOf[Sin],
+      classOf[Size],
+      classOf[SortArray],
+      classOf[SoundEx],
+      classOf[SparkPartitionID],
+      classOf[Sqrt],
+      classOf[Stack],
+      classOf[StringInstr],
+      classOf[StringLPad],
+      classOf[StringRPad],
+      classOf[StringRepeat],
+      classOf[StringReplace],
+      classOf[StringToMap],
+      classOf[StringTrim],
+      classOf[StringTrimLeft],
+      classOf[StringTrimRight],
+      classOf[Substring],
+      classOf[SubstringIndex],
+      classOf[Tan],
+      classOf[Tanh],
+      classOf[ToDegrees],
+      classOf[ToRadians],
+      classOf[ToUnixTimestamp],
+      classOf[UnaryPositive],
+      classOf[Unhex],
+      classOf[UnixMicros],
+      classOf[UnixMillis],
+      classOf[UnixSeconds],
+      classOf[Upper],
+      classOf[Uuid],
+      classOf[WeekDay],
+      classOf[WeekOfYear],
+      classOf[WidthBucket],
+      classOf[Year],
+      classOf[ZipWith]
+    )
+
+    // Some functions are fully supported with ANSI mode off
+    lazy val ansiOff = Seq(
+      classOf[Remainder],
+      classOf[Multiply],
+      classOf[Add],
+      classOf[Subtract],
+      classOf[Divide],
+      classOf[Abs],
+      classOf[Pmod]
+    )
+
+    if (SQLConf.get.ansiEnabled) {
+      ansiOn
+    } else {
+      ansiOn ++ ansiOff
+    }
+  }
+
+  def canBePushedToHybrid(child: SparkPlan, conf: RapidsConf): String = {
+    child match {
+      case fsse: FileSourceScanExec if HybridFileSourceScanExecMeta.useHybridScan(conf, fsse) =>
+        conf.pushDownFiltersToHybrid
+      case _ => "OFF"
+    }
+  }
+
+  def recursivelySupportsHybridFilters(condition: Expression): Boolean = {
+    condition match {
+      case filter if HybridExecutionUtils.supportedByHybridFilters
+          .exists(_.isInstance(filter)) =>
+        val childrenSupported = filter.children.forall(recursivelySupportsHybridFilters)
+        childrenSupported
+      case _ => false
+    }
+  }
+
+  /** 
+   * Search the plan for FilterExec whose child is a FileSourceScanExec if Hybrid Scan is enabled
+   * and decide whether to push down a filter to the GPU or not according to whether CPU/GPU
+   * support it. After that we can remove the condition from one side to avoid duplicate execution
+   * or unnecessary fallback/crash.
+   */
+  def applyHybridScanRules(plan: SparkPlan, conf: RapidsConf): SparkPlan = {
```

Comment:
Rename to `tryToApplyHybridScanRules`?
And at the first line of this function,  check if Hybrid feature is enabled to avoid executing the following code.